wilson/run/smeft/smpar.py

```diff
@@ -37,45 +37,44 @@
 
 def m2Lambda_to_vMh2(m2, Lambda, C):
     """Function to numerically determine the  physical Higgs VEV and mass
-    given the parameters of the Higgs potential."""
-    try:
-        v = (sqrt(2 * m2 / Lambda) + 3 * m2**(3 / 2) /
-             (sqrt(2) * Lambda**(5 / 2)) * C['phi'].real)
-    except ValueError:
-        v = 0
-    Mh2 = 2 * m2 * (1 - m2 / Lambda * (3 * C['phi'].real - 4 * Lambda * C['phiBox'].real +
-                                        Lambda * C['phiD'].real))
-    return {'v': v, 'Mh2': Mh2}
+    given the parameters of the Higgs potential.
 
-def _vMh2_to_m2Lambda_SM(v, Mh2):
-    m2 = Mh2/2
-    Lambda = 2 * m2 / v**2
-    return {'m2': m2, 'Lambda': Lambda}
+    The relations used by this function have been derived from the Lagrangian
+    in eq. (3.1) of arXiv:1704.03888. This paper presents results expanded up to
+    linear order in the Wilson coefficients while the all-order expressions are
+    implemented here.
+    """
+    if not m2 > 0 or not Lambda > 0:
+        raise ValueError('`m2` and `Lambda` are expected to be positive.')
+    Cphi = C['phi'].real
+    Ckin = C['phiBox'].real - C['phiD'].real / 4
+    if abs(Cphi) < 1e-16:
+        v2 = 2 * m2 / Lambda
+    else:
+        sqrt_arg = Lambda**2 - 12 * Cphi * m2
+        if not sqrt_arg >= 0:
+            raise ValueError("'Lambda**2 - 12 * Cphi * m2' must be positive.")
+        v2 = ( Lambda - sqrt(sqrt_arg) )/( 3 * Cphi )
+    Mh2 = v2 * ( 1 + Ckin * v2 )**2 * ( Lambda - 3 * Cphi * v2 )
```

Comment:
I get here for the second factor: ( 1 - Ckin * v2 )**-2

Reply:
@jasonaebischerGIT looking at it again, I indeed get a different result than I wrote there. But if I normalize the Higgs kinetic term without performing an expansion in the Wilson coefficients, I actually get (compared to (3.4) - (3.6) in 1704.03888)
```
H = h * (1 - 2 * Ckin v2 )**(-1/2)
```
I had used this result expanded to linear order in Ckin, which gives `H = h * (1 + Ckin v2)` or `h = H * (1 - Ckin * v2)` (I think I used the former and you used the latter). However, if I don't expand to linear order in Ckin, I actually get
```
Mh2 = v2 / (1 - 2 * Ckin * v2) * (Lambda - 3 * Cphi * v2)
```
If you agree, I will change this line (and also the inverse relation used in `vMh2_to_m2Lambda`)

Reply:
@peterstangl Thanks for checking again. If I don't expand I obtain the same result as you for the renormalization factor Zh of the Higgs field. I think however, that one has to do an expansion there, even though the renormalization is finite, since the renormalization relies on an order-by-order expansion, in this case up to dim6. Therefore I would say that one has use the Zh from 1704.03888, entering the expression for Mh.

Reply:
@jasonaebischerGIT I do not understand why one *has* to do an expansion there. Isn't this just a rescaling of the field that should make the kinetic term canonically normalized? If I do not expand, the kinetic term is exactly canonically normalized in the presence of the dim6 operators (but of course it might not be canonically normalized anymore in the presence of higher dimensional operators). If I expand, the kinetic term is only canonically normalized up to additional terms of order v^4 that are neglected. So for this scaling factor, the difference between not expanding and expanding is always of higher order. Why should it then be a problem to make the kinetic term exactly canonically normalized in the absence of higher dimensional operators?

Reply:
@peterstangl What bothers me is that one does not get the same if one takes your expression and expands it up to dim6, compared to expanding from the beginning. This means that even the dim6 term contributing to the Higgs mass does not come out right in one of the two ways.
Also, taking into account products of WCs, corresponding to dim8 contributions is certainly not completely consistent, but I guess one can neglect this effect issue here.

Reply:
@jasonaebischerGIT I think I don't agree with what you get. If I take my result not using the expansion, i.e.
```
Mh2 = v2 / (1 - 2 * Ckin * v2) * (Lambda - 3 * Cphi * v2)
```
and then I expand this result in `v2` up to order v^4, then I get exactly the result shown in 1704.03888 in the second line of eq. (3.8). So up to order v^4, the two results are exactly equal and differences only appear at v^6. But v^6 we are neglecting anyway, so I would think it is fine to use either of them for `Mh2`

Reply:
@peterstangl As far as I can see we only differ in the second factor appearing in Mh2, which for you is 

`1/ (1 - 2 * Ckin * v2)`

and for me 

`( 1 - Ckin * v2 )**-2`.

Expanding both of these factors gives the same up to v2, namely:

`1 +2 Ckin * v2 `

Therefore both results coincide with the results in 1704.03888 and since they are the same up to v2^2 we can use either of them I suppose. Do you agree?

Reply:
@jasonaebischerGIT yes exactly, I fully agree! Since the other parts of the relations are not expanded in v, I would then just use the unexpanded expression also there.

Reply:
@peterstangl Very good, time to finally merge the PR!